codec-http2/src/main/java/io/netty/handler/codec/http2/Http2Connection.java

```diff
@@ -257,11 +256,22 @@
         int maxActiveStreams();
 
         /**
-         * Sets the maximum number of streams (created by this endpoint) that are allowed to be active at once.
-         * This is the {@code SETTINGS_MAX_CONCURRENT_STREAMS} value sent from the opposite endpoint to
+         * The limit imposed by {@link #maxActiveStreams()} does not apply to streams in the IDLE state. Since IDLE
+         * streams can still consume resources this limit will include streams in all states.
+         * @return The maximum number of streams that can exist at any given time.
+         */
+        int maxStreams();
+
+        /**
+         * Sets the limit for {@code SETTINGS_MAX_CONCURRENT_STREAMS} and the limit for {@link #maxStreams()}.
+         * @param maxActiveStreams The maximum number of streams (created by this endpoint) that are allowed to be
+         * active at once. This is the {@code SETTINGS_MAX_CONCURRENT_STREAMS} value sent from the opposite endpoint to
          * restrict stream creation by this endpoint.
+         * @param maxStreams The limit imposed by {@link #maxActiveStreams()} does not apply to streams in the IDLE
+         * state. Since IDLE streams can still consume resources this limit will include streams in all states.
+         * @throws Http2Exception if {@code maxStreams < maxActiveStream}.
          */
-        void maxActiveStreams(int maxActiveStreams);
+        void maxStreams(int maxActiveStreams, int maxStreams) throws Http2Exception;
```

Comment:
I don't think most user of the codec will call these APIs. If these APIs are called outside of the SETTINGS exchange there is a potential the SETTINGS with the remote peer will get out of sync. If we add this method back and deprecated it ... are we thinking using the deprecated method in `DefaultHttp2ConnectionEncoder` is preferable? Seems like this will just require additional code, duplicating conditionals, and a wider API for minimal gain. WDYT @normanmaurer @nmittler ?